### PR TITLE
GitHub Search React/week4

### PR DESCRIPTION
**Pull request by shanhaider23**

### CODE REVIEW

File: react/week4/github-search/src/components/ProfileSearch.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      setUsers(data.items);
+    } catch (error) {
+      setIsLoading(true);
+      setIsError(error);
+    }
+  };
+
+  useEffect(() => {
+    gitHubApiHandler();
+  }, []);
```
Comment (calm-prog) [inline react/week4/github-search/src/components/ProfileSearch.js]: (this is the continuation of another comment)

... here you can pass any state into the dependency array, eg.:
```
useEffect(() => {
    gitHubApiHandler();
  }, [input]);
```

In this case every time the input state changes, then the function inside the useEffect will be triggered.

File: react/week4/github-search/src/components/InputForm.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React, { useContext } from "react";
+import { ProfileContext } from "./ProfileSearch";
+import DisplayUser from "./DisplayUsers";
+
+function Input() {
+  const { gitHubApiHandler } = useContext(ProfileContext);
+  const onChangeHandler = (event) => {
+    gitHubApiHandler(event.target.value);
```
Comment (calm-prog) [inline react/week4/github-search/src/components/InputForm.js:8]: It's a best practice to call side effects like fetch inside the useEffect method. 

I would rather create an input state and pass it into the useEffect dependency array...

**Review by calm-prog — COMMENTED** [review]
Super good. Very detailed and thorough in the logic, and it's also great how you use modern syntax, eg async - await.

I have only one comment.

File: react/week4/github-search/src/components/ProfileSearch.js (hunk 1 above)
Comment (shanhaider23) [reply react/week4/github-search/src/components/ProfileSearch.js]: Thanks 👍 I will change it

**Pull request merged**